sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/ParameterAssignedTo.CSharp9.cs

```diff
@@ -0,0 +1,21 @@
+﻿using System;
+
+static void Foo(int x)
+{
+    Action<int, int> discard = (_, _) => _ = 10; // Compliant
+
+    Action<int, int> foo = (a, b) => a = 10; // Noncompliant
+}
+
+public record Record
```

Comment:
how does this rule behave with `set` and `init` for Properties, if `value` is assigned?
IMO it should raise

Reply:
Nice! I've added the test cases. 

It works as expected but I did not think at all that you might assign to `value`.